### ISSUE TITLE
Loss problems

### ISSUE THREAD

**Issue by Youssef-Barakat** [SWivid/F5-TTS#972]
### Checks

- [x] This template is only for research question, not usage problems, feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I am using English to submit this issue to facilitate community communication.

### Question details

After training my TTS model, I've observed that while the output WAV files have improved, the model exhibits overfitting behaviors:​

Overgeneration: The model tends to generate certain words even when they're not present in the input text.​

Loss Oscillation: Throughout training, the loss fluctuates without a clear downward trend, indicating instability.

do you know that cause of this?

**Comment by SWivid** [comment]
> Loss Oscillation: Throughout training, the loss fluctuates without a clear downward trend, indicating instability.

#406 #9 etc.

> Overgeneration: The model tends to generate certain words even when they're not present in the input text.​

samely, check readme and existing issue for inference

**Comment by Youssef-Barakat** [comment]
This isn’t the same as the issues mentioned, since I'm fine-tuning the model—not training it from scratch—and the loss doesn’t show smooth or consistent improvement.

**Comment by SWivid** [comment]
got it, that's why we mentioned that `This template is only for research question, not usage problems, feature requests or bug reports.` and:

![Image](https://github.com/user-attachments/assets/1388384c-337c-4593-ae7b-6095fe47972c)

also, how is your loss value in specific, any curve screenshots? dataset amount and language type?

**Comment by Youssef-Barakat** [comment]
![Image](https://github.com/user-attachments/assets/84effe51-f7ae-4f6e-bd87-690514885198)
dataset ~ 250 hours

**Comment by SWivid** [comment]
so same language or not? how is the effective batchsize

**Comment by Youssef-Barakat** [comment]
not the same language it's a new one, batch_size_per_gpu is 4000, dataset supposed to be clean denoised

**Comment by SWivid** [comment]
if a new language, your loss curve doesn't make sense. if sft with v1_base/base on unseen language, the loss will not start from 0.6

> Loss Oscillation: Throughout training, the loss fluctuates without a clear downward trend, indicating instability.
This isn’t the same as the issues mentioned, since I'm fine-tuning the model—not training it from scratch—and the loss doesn’t show smooth or consistent improvement.

then it is just the same as previously mentioned issues, from what point do you think the loss has strange behaviour?

**Comment by Youssef-Barakat** [comment]
this is what makes it strange as it started from .6 and it's still the same while training a new language.

**Comment by SWivid** [comment]
what is the dataset you used? please provide config yaml or the prepare data interface screenshots

**Comment by Youssef-Barakat** [comment]
![Image](https://github.com/user-attachments/assets/88ed8471-36e0-45dc-a562-b3a628bf6013)
It's not an open source dataset

**Comment by SWivid** [comment]
no idea then.
apart from loss starting from 0.6, thought the curve is just fine cuz as mentioned in #406 mel loss is just like that.

for overfitting problem, using more data with more diversity will help if you want better zero-shot ability.
for word leakage problem, check existing issues: make sure you inference operation is of best practice as mentioned in readme and the length is not exceeding the maxium during your sft process

**Comment by Youssef-Barakat** [comment]
okay, Thank you for your help and your great work!
